File: editorial/chap3-3/prob3-3-3.cpp
#include <iostream>
using namespace std;

long long n, r;
long long Fact_n = 1;
long long Fact_r = 1;
long long Fact_nr = 1;

int main() {
	// 入力
	cin >> n >> r;
	
	// 階乗の計算
	for (int i = 1; i <= n; i++) Fact_n *= i;
	for (int i = 1; i <= r; i++) Fact_r *= i;
	for (int i = 1; i <= n - r; i++) Fact_nr *= i;
	
	// 出力
	cout << Fact_n / (Fact_r * Fact_nr) << endl;
	return 0;
}

File: editorial/chap3-3/prob3-3-3.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		// 入力
		int n = sc.nextInt();
		int r = sc.nextInt();
		
		// 会場の計算
		long fact_n = 1, fact_r = 1, fact_nr = 1;
		for (int i = 1; i <= n; i++) fact_n *= i;
		for (int i = 1; i <= r; i++) fact_r *= i;
		for (int i = 1; i <= n - r; i++) fact_nr *= i;
		
		// 出力
		System.out.println(fact_n / (fact_r * fact_nr));
	}
}